maven-embedder/src/main/java/org/apache/maven/cli/transfer/SimplexTransferListener.java

```diff
@@ -0,0 +1,219 @@
+        put(event, false);
+    }
+
+    @Override
+    public void transferStarted(TransferEvent event) {
+        put(event, false);
+    }
+
+    @Override
+    public void transferProgressed(TransferEvent event) {
+        put(event, false);
+    }
+
+    @Override
+    public void transferCorrupted(TransferEvent event) {
+        put(event, false);
+    }
+
+    @Override
+    public void transferSucceeded(TransferEvent event) {
+        ongoing.remove(event.getResource().getFile());
+        put(event, ongoing.isEmpty());
+    }
+
+    @Override
+    public void transferFailed(TransferEvent event) {
+        ongoing.remove(event.getResource().getFile());
+        put(event, ongoing.isEmpty());
+    }
+
+    private static class Exchange {
+        private final TransferEvent event;
+
+        private Exchange(TransferEvent event) {
+            this.event = event;
+        }
+
+        public void process(Consumer<TransferEvent> consumer) {
+            consumer.accept(event);
+        }
+
+        public void waitForProcessed() throws InterruptedException {
+            // nothing, is async
+        }
+    }
+
+    private static class BlockingExchange extends Exchange {
+        private final CountDownLatch latch = new CountDownLatch(1);
+
+        private BlockingExchange(TransferEvent event) {
+            super(event);
+        }
+
+        @Override
+        public void process(Consumer<TransferEvent> consumer) {
+            super.process(consumer);
+            latch.countDown();
+        }
+
+        @Override
+        public void waitForProcessed() throws InterruptedException {
+            latch.await();
+        }
+    }
+
+    private void demux(List<Exchange> exchanges) {
+        for (Exchange exchange : exchanges) {
+            exchange.process(transferEvent -> {
+                TransferEvent.EventType type = transferEvent.getType();
+                try {
+                    switch (type) {
+                        case INITIATED:
+                            delegate.transferInitiated(transferEvent);
+                            break;
+                        case STARTED:
+                            delegate.transferStarted(transferEvent);
+                            break;
+                        case PROGRESSED:
+                            delegate.transferProgressed(transferEvent);
+                            break;
+                        case CORRUPTED:
+                            delegate.transferCorrupted(transferEvent);
+                            break;
+                        case SUCCEEDED:
+                            delegate.transferSucceeded(transferEvent);
+                            break;
+                        case FAILED:
+                            delegate.transferFailed(transferEvent);
+                            break;
+                        default:
+                            throw new IllegalArgumentException("Type: " + type);
```

Comment:
doing this does not make any sense IMO, maybe just ignore it? It will just kill the daemon thread, killing any transport logging of Maven...